src/test/java/io/pinecone/clients/ConnectionsMapTest.java

```diff
@@ -0,0 +1,118 @@
+package io.pinecone.clients;
+
+import io.pinecone.configs.PineconeConfig;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.openapitools.client.ApiException;
+import org.openapitools.client.api.ManageIndexesApi;
+import org.openapitools.client.model.IndexModel;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ConnectionsMapTest {
+
```

Comment:
We should also try to add some tests around the connection being re-used. In other words, if someone calls `getIndexConnection("my-index")` multiple times, they should get the same instance back each time.

Reply:
Updated the tests for number of connections created and in future, we will like to test for the contents as well